### PR TITLE
Cleanup cmake.eclass EAPI-7 stragglers

### PR DESCRIPTION
**Pull request by a17r**
In the year 2024 we do not bump new ebuilds as EAPI-7. **Please.**

### CODE REVIEW

**Comment by beatussum** [comment]
https://github.com/gentoo/gentoo/pull/35651/commits/b7ac755fe9a712af29bd11d7d5892a656e8b4ac1 is duplicate with https://github.com/gentoo/gentoo/pull/35603.

**Review by Chiitoo — APPROVED** [review]
Should be good to go with dropping 'media-gfx/nomacs-3.16.224-r5', with r6 being stable and all.

Thanks!

**Comment by mattst88** [comment]
Ack for [dev-libs/cJSON: drop 1.7.16](https://github.com/gentoo/gentoo/pull/35651/commits/25992405ac6d9a61a5a3ba1e5543ff0bc42512a2)

**Review by chewi — APPROVED** [review]
Cool for games.

**Review by ZeroChaos- — APPROVED** [review]

File: gnustep-base/libobjc2/libobjc2-2.1.ebuild [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-RESTRICT="!test? ( test )"
-
-RDEPEND="boehm-gc? ( dev-libs/boehm-gc )"
-BDEPEND="${RDEPEND}
-	sys-devel/clang"
-
-PATCHES=(
-	"${FILESDIR}"/${P}-eh_trampoline.patch
-	"${FILESDIR}"/${P}-pthread_link.patch
-)
```
Comment (voyageur) [inline gnustep-base/libobjc2/libobjc2-2.1.ebuild:25]: These patches are not used in 2.2 so can be removed too

**Review by voyageur — CHANGES_REQUESTED** [review]
Current libobjc2 2.2 looks good and has up-to-date EAPI, just note on patches cleanup

File: gnustep-base/libobjc2/libobjc2-2.1.ebuild (hunk 1 above)
Comment (a17r) [reply gnustep-base/libobjc2/libobjc2-2.1.ebuild:25]: thx

**Comment by atupone** [comment]
Personally I'm OK to remove the old glm version,  but I have not done because is stable and I have not tested all the reverse deps.
Actually if one of these fails during build is just one liner fix:
add
#define GLM_ENABLE_EXPERIMENTAL
at the begin of each failing source.

**Review by atupone — APPROVED** [review]
Personally I'm OK to remove the old glm version, but I have not done because is stable and I have not tested all the reverse deps.
Actually if one of these fails during build is just one liner fix:
add
#define GLM_ENABLE_EXPERIMENTAL
at the begin of each failing source.

**Comment by a17r** [comment]
> Personally I'm OK to remove the old glm version, but I have not done because is stable and I have not tested all the reverse deps. Actually if one of these fails during build is just one liner fix: add #define GLM_ENABLE_EXPERIMENTAL at the begin of each failing source.

The only reported failure with that was recently fixed and is currently pending stabilisation, I'm not going to drop the old version before that is done: Personally I'm OK to remove the old glm version, but I have not done because is stable and I have not tested all the reverse deps. Actually if one of these fails during build is just one liner fix: add #define GLM_ENABLE_EXPERIMENTAL at the begin of each failing source.

**Pull request merged**